Makefile

```diff
@@ -0,0 +1,9 @@
+build: build-heroku-16 build-heroku-20
+
+build-heroku-16:
+	@echo "Building imagemagick in Docker for heroku-16..."
+	@docker run -v $(shell pwd):/buildpack --rm -it -e "STACK=heroku-16" -w /buildpack heroku/heroku:16-build scripts/build_imagemagick imagemagick-heroku-16.tar.gz
```

Comment:
`-v` we set the volume with the current directory and `/buildpack` is where the volume will be mounted inside the container.

`-e` we pass the `STACK` variable to the `build` script.

`heroku/heroku:16-build` is the image name

`scripts/build_imagemagick` is the script to be executed

`imagemagick-heroku-16.tar.gz` the last parameter is used by the `build` script to rename the compiled file.

Reply:
Wow, thanks for explaining all of this. All these arguments were not so clear to me, but they are now!